docs/website/docs/general-usage/transformations/index.md

```diff
@@ -0,0 +1,197 @@
+---
+title: Transforming loaded datasets
+description: Define SQL-based or mixed SQL + Python transformations on data that is **already** in your destination.
+keywords: [transformation, dataset, sql, pipeline, ibis, arrow]
+---
+
+# Transformations: reshape data you have already loaded
+
+:::danger
+The `dlt.transformation` decorator is currently in preview and while it may appear to work very well, it should not be considered stable. If you found this unlinked page and would like to give us feedback, please let us know in our Slack community.
+:::
+
+`dlt transformations` let you build new tables or views from datasets that have _already_ been ingested with `dlt`.  
+Instead of pulling from an external API, remote database, or custom source like `@dlt.resource` does, a transformation's **input** is a `dlt.Dataset`, and its **output** is a source
+that can be loaded into the same dataset as the input, another dataset on the same destination, or even to another destination on another database. Depending on the locations of the input and
+output datasets, these transformations will be executed in SQL, or just the query is executed as SQL, extracted as Arrow tables, and then materialized in the destination dataset in the same
+way that regular dlt resources are.
+
+You create them with the `@dlt.transformation` decorator which has the same signature as the `@dlt.resource` decorator, but does not yield items but rather a SQL query including the resulting
+column schema. dlt transformations support the same write_dispositions per destination as dlt resources do.
+
+## Motivations
+
+A few real-world scenarios where dlt transformations can be useful:
+
+- **Build one-stop reporting tables** – Flatten and enrich raw data into a wide table that analysts can pivot, slice, and dice without writing SQL each time.  
+- **Normalize JSON into 3-NF** – Break out repeating attributes from nested JSON so updates are consistent and storage isn't wasted.  
+- **Create dimensional (star-schema) models** – Produce fact and dimension tables so BI users can drag-and-drop metrics and break them down by any dimension.  
+- **Generate task-specific feature sets** – Deliver slim tables tailored for personalization, forecasting, or other ML workflows.  
+- **Apply shared business definitions** – Encode rules such as "a *sale* is a transaction whose status became *paid* this month," ensuring every metric is counted the same way.  
```

Comment:
this is interesting. can transformations share something? ie expression that we name `is_sale`? how do you see this implemented? potentially this is powerful because you can define rules in code, not some yaml like DSL

Reply:
I got this from @adrianbr  :)